packages/plugin/src/VisualizationPlugin.js

```diff
@@ -198,37 +205,101 @@ export const VisualizationPlugin = ({
         ? { getBoundingClientRect: () => contextualMenuRect }
         : null
 
+    const hasLegendSet = fetchResult.legendSets?.length > 0
+    const transformedStyle =
+        forDashboard && hasLegendSet
+            ? {
+                  ...style,
+                  width: style.width - (showLegendKey ? 200 : 36), // TODO: Very arbitrary and static, change these values to something dynamic
+              }
+            : style
+
+    const getLegendKey = () => {
```

Comment:
Perhaps we can move this into a separate file (like we do with `ContextualMenu`)?

Reply:
Good point, this bloats VisualizationPlugin.js and should ideally be moved. I tried moving it, but it created an issue with the width calculation of the chart, which pushes the legend key out of bounds. Let's keep it as is for now and revisit this in a future refactor session.